functional_tests/scylla_operator/test_functional.py

```diff
@@ -87,6 +91,130 @@ def test_single_operator_image_tag_is_everywhere(db_cluster):
+
+        # NOTE: check that Scylla services are headless - no IPs are set
+        svc_ips = kubectl(
+            "get svc --no-headers -o=custom-columns=':.spec.clusterIP'"
+            f" -l scylla/cluster={cluster_name} -l scylla-operator.scylladb.com/scylla-service-type=member",
+            namespace=namespace).stdout.split()
+        assert svc_ips
+        assert len(svc_ips) == 3
+        assert all(svc_ip in ('', 'None') for svc_ip in svc_ips), "SVC IPs were expected to be absent"
+
+        # NOTE: read Scylla pods IPs
+        pods_data = get_pod_names_and_ips(cluster_name=cluster_name, namespace=namespace)
+        try:
+            db_cluster.k8s_cluster.create_namespace(namespace=namespace2)
+            db_cluster.k8s_cluster.create_scylla_manager_agent_config(namespace=namespace2)
+
+            log.info('Deploy second ScyllaCluster')
+            values.set("datacenter", f"{dc_name}-2")
+            values.set("externalSeeds", list(pods_data[namespace].values()))
+            values.set("fullnameOverride", cluster_name2)
+            log.debug(db_cluster.k8s_cluster.helm_install(
+                target_chart_name=target_chart_name2, source_chart_name="scylla-operator/scylla",
+                version=operator_version, use_devel=True, values=values, namespace=namespace2))
+            db_cluster.k8s_cluster.kubectl_wait(
+                "--all --for=condition=Ready pod", namespace=namespace2, timeout=1200)
+            pods_data |= get_pod_names_and_ips(cluster_name=cluster_name2, namespace=namespace2)
+            ip_to_dc_map = {}
+            for current_namespace in pods_data.keys():
+                for pod_ip in pods_data[current_namespace].values():
+                    ip_to_dc_map[pod_ip] = dc_name + "" if current_namespace == namespace else "-2"
+
+            log.info("Verify DB cluster's peers info")
+            for current_namespace in pods_data.keys():
+                for pod_name, pod_ip in pods_data[current_namespace].items():
+                    cqlsh_cmd = "SELECT JSON peer, data_center, rack, rpc_address FROM system.peers;"
+                    cqlsh_results = kubectl(
+                        f"exec {pod_name} -- /bin/cqlsh -e \"{cqlsh_cmd}\"",
+                        namespace=current_namespace).stdout.split("---\n")[-1].split("\n")
+                    table_rows = [yaml.safe_load(row) for row in cqlsh_results if "{" in row]
+                    assert len(table_rows) == 5, "Expected 5 peers"
+                    for row in table_rows:
+                        assert row["peer"] == row["rpc_address"]
+                        assert row["peer"] != pod_ip
+                        assert row["peer"] in ip_to_dc_map
+                        assert row["rack"] == db_cluster.k8s_cluster.params.get('k8s_scylla_rack')
+                        assert row["data_center"] == ip_to_dc_map[row["peer"]]
+            need_to_collect_logs = False
+        finally:
+            if need_to_collect_logs:
```

Comment:
I would have put the cleanups into a test finalizers instead of try/finally , but it's not that critical

Reply:
Pytest has bug where finalizers don't have strict order: https://github.com/pytest-dev/pytest/issues/8489
So, it doesn't fit our case. We should, at first, gather logs, and only then delete resources. Not vice-versa.

Reply:
why do you need ordering ?, from the looks of it you are collecting logs from two different scylla cluster
why do you care who will be first and who will be last ?

Reply:
Ordering between logs gathering and cleanup, both will need to go to finalizers.